_assignments/mp1.md

```diff
@@ -0,0 +1,395 @@
+Pressing it repeatedly should change the color each time, i.e. have it be a toggle or a switch between a series of colors.
+Make sure that the color change is large enough so it is immediately apparent!
+
+Create a new script. To attach it to the light object, select the light, then drag the script from the "Assets" tab to the Inspector tab on the right.
+When a Unity script is attached to a GameObject, that script will run when the game is started.
+Furthermore, the `this` reference in the script will refer to the object that the script is attached to.
+Our first step is to get the `Light` component of our point light `GameObject`.
+Read the [Controlling GameObjects using GetComponent](https://docs.unity3d.com/Manual/ControllingGameObjectsComponents.html) tutorial, then add these lines to your script:
+
+![Inititalize Light](/img/assignments/mp1/image9.png)
+
+This will set the `light` variable to reference the `Light` component of the object this script is attached to when we initially start the game. N.b.: calling `GetComponent<>()` is the same as calling `this.GetComponent<>()`.
+To register input, we use the [Input](https://docs.unity3d.com/ScriptReference/Input.html) library of Unity,
+specifically, the `Input.GetKeyDown` method, which will return `true` when the specified key is first pressed down.
+Since we want to listen for the `Tab` key, then in our `Update` method, write:
+
+![Get Tab Key Down](/img/assignments/mp1/image14.png)
+
+For the actual light color, you can either create a new color using the `new Color(red, green, blue)` constructor, or one of the predefined colors. How you change the light is up to you, but the light should visibly change every time we press `Tab`.
+
+### Orbiting Moon
+
+> The moon should orbit the planet sphere.
+The easiest way to do this is to have the planet constantly rotate.
+Since the moon is a child of the planet, it will also rotate around the planet.
+
+You control the rotation and position of a `GameObject` with the [`Transform`](https://docs.unity3d.com/ScriptReference/Transform.html) class, using the `<GameObjectName>.transform` object variable.
+This class is well worth looking through,
+though the most important parts for this MP are the `transform.position`, the vector of the object's global coordinates, and the `Rotate` method.
+
+Most of Unity rotations are done using something called quaternions, which are better than the standard way of measuring rotation about the $$x$$, $$y$$, and $$z$$ axes.
+You will learn about quaternions, and why they are awesome, in class later.
+For now, simply know that the `transform.Rotate(Vector3(a, b, c))` will rotate object `a` degrees about the object’s x axis, `b` degrees about the y, and `c` degrees about the z.
+Create a script, called “orbit”, and attach it to the parent sphere. In its update method, add this line:
+
+```
+this.transform.Rotate(new Vector3(0, 60 * Time.deltaTime, 0));

```

Comment:
the use of an image here (and any other complete lines of code) was intentional -- this required students to actually type out the code, rather than copy-pasting it. I'd recommend using that.